overlay/src/components/organisms/widgets/TeamView.js

```diff
@@ -281,16 +290,47 @@ const TeamVideo = ({ teamId, type, setIsLoaded }) => {
 };
 
 export const TeamView = ({ widgetData: { settings }, transitionState }) => {
-    const [isLoaded, setIsLoaded] = useState(settings?.mediaType === undefined);
+    const dispatch = useDispatch();
+    const [loadedComponents, setLoadedComponents] = useState(0);
+    const isLoaded = loadedComponents === (1 << settings.content.length) - 1;
     return <TeamViewContainer
         show={isLoaded}
         animation={isLoaded && (transitionState === "exiting" ? slideOut : slideIn)}
         animationStyle={transitionState === "exiting" ? "ease-in" : "ease-out"}
     >
-        {settings.mediaType !== undefined &&
-            <TeamVideo teamId={settings.teamId} type={settings.mediaType} setIsLoaded={setIsLoaded}/>
-        }
-        {settings.showTaskStatus !== false && <ScoreboardColumn teamId={settings.teamId}/>}
+        {settings.content.map((c, index) => {
+            const setIsLoaded = () => setLoadedComponents(m => m | (1 << index));
+            if (c.type === "TaskStatus") {
+                useEffect(() => {
+                    setIsLoaded(true);
+                }, [c.teamId]);
+                return <ScoreboardColumn teamId={c.teamId}/>;
+            }
+            if (c.type === "Photo") {
+                return <TeamVideoAnimationWrapper>
+                    <TeamImageWrapper src={c.url} onLoad={() => setIsLoaded(true)}/>
+                </TeamVideoAnimationWrapper>;
+            }
+            if (c.type === "Video") {
+                return <TeamVideoAnimationWrapper>
+                    <TeamVideoWrapper
+                        src={c.url}
+                        onCanPlay={() => setIsLoaded(true)}
+                        onError={() => setIsLoaded(false) || dispatch(pushLog("ERROR on loading video in TeamView widget"))}
```

Comment:
setIsLoaded is a function with no arguments.
the argument is ignored, so this won't change what's displayed